Xamarin.Essentials/Types/Calendar.shared.cs

```diff
@@ -43,32 +47,138 @@ public CalendarEvent(string id, string calendarId, string title)
+        }
 
-        public TimeSpan Duration =>
-            AllDay ? TimeSpan.FromDays(1) : EndDate - StartDate;
+        public string Url { get; set; }
 
         public IEnumerable<CalendarEventAttendee> Attendees { get; set; }
+
+        public IEnumerable<CalendarEventReminder> Reminders { get; set; }
+
+        public RecurrenceRule RecurrancePattern { get; set; }
     }
 
+    [Preserve(AllMembers = true)]
     public class CalendarEventAttendee
     {
-        public CalendarEventAttendee()
-        {
-        }
-
-        public CalendarEventAttendee(string name, string email)
-        {
-            Name = name;
-            Email = email;
-        }
-
         public string Name { get; set; }
 
         public string Email { get; set; }
+
+        public AttendeeType Type { get; set; }
+
+        public bool IsOrganizer { get; set; }
+    }
+
+    [Preserve(AllMembers = true)]
+    public class CalendarEventReminder
+    {
+        public int MinutesPriorToEventStart { get; set; }
+    }
+
+    [Preserve(AllMembers = true)]
+    public class RecurrenceRule
+    {
+        public uint? TotalOccurrences { get; set; }
+
+        public uint Interval { get; set; }
+
+        public DateTimeOffset? EndDate { get; set; }
+
+        public RecurrenceFrequency? Frequency { get; set; }
+
+        // Only allow event to occur on these days [not available for daily]
+        public List<CalendarDayOfWeek> DaysOfTheWeek { get; set; }
+
+        public uint DayOfTheMonth { get; set; }
+
+        public MonthOfYear? MonthOfTheYear { get; set; }
+
+        public IterationOffset? WeekOfMonth { get; set; }
+    }
+
+    public enum RecurrenceFrequency
+    {
+        None = -1,
+        Daily = 0,
+        Weekly = 1,
+        Monthly = 2,
+        MonthlyOnDay = 3,
+        Yearly = 4,
+        YearlyOnDay = 5
+    }
+
+    public enum CalendarDayOfWeek
+    {
+        None = 0,
+        Sunday = 1,
+        Monday = 2,
+        Tuesday = 4,
+        Wednesday = 8,
+        Thursday = 16,
+        Friday = 32,
+        Saturday = 64,
+        Weekday = Monday | Tuesday | Wednesday | Thursday | Friday,
+        Weekend = Saturday | Sunday,
+        AllDays = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday,
+    }
+
+    public enum MonthOfYear
+    {
+        January = 1,
+        February = 2,
+        March = 3,
+        April = 4,
+        May = 5,
+        June = 6,
+        July = 7,
+        August = 8,
+        September = 9,
+        October = 10,
+        November = 11,
+        December = 12
+    }
+
+#if __ANDROID__ || __IOS__
+    public enum IterationOffset
+    {
+        Last = -1,
+        First = 1,
+        Second = 2,
+        Third = 3,
+        Fourth = 4
+    }
+
+#else
+    public enum IterationOffset
+    {
+        First = 0,
+        Second = 1,
+        Third = 2,
+        Fourth = 3,
+        Last = 4
+    }
+
+#endif
```

Comment:
Because this is shared code, and these are constants, we cannot have different values. If I write netstandard and pick `First`, it will use the value `0` - which does not match anything on Android.